### PR TITLE
Fix more clippy warnings

### PR DESCRIPTION
**Pull request by zecakeh**
Follow-up to  #3291.

### CODE REVIEW

File: .clippy.toml [hunk 1]
```diff
@@ -0,0 +1 @@
+assigning_clones = false
```
Comment (poljar) [inline .clippy.toml:1]: Missing newline at the end of the file.

**Review by poljar — CHANGES_REQUESTED** [review]
Looks good, apart from the missing newline.

File: .clippy.toml (hunk 1 above)
Comment (zecakeh) [reply .clippy.toml:1]: Yeah that doesn't work (that's why CI is failing). I don't think we can disable lints at the workspace level; I'll have to add it to each crate.

**Review by poljar — APPROVED** [review]
Despite being a bit annoying, seems to work. Thanks.

**Pull request merged**